chalicelib_smaht/checks/helpers/google_utils.py

```diff
@@ -0,0 +1,1107 @@
+            report_request_count = len(formatted_report_requests)
+            if report_request_count > 5:
+                raw_result = { "reports" : [] }
+                for chunk_num in range(report_request_count // 5 + 1):
+                    chunk_num_start = chunk_num * 5
+                    chunk_num_end = min([chunk_num_start + 5, report_request_count])
+                    if chunk_num_start < chunk_num_end:
+                        for chunk_raw_res in self._api.batch_run_reports(BatchRunReportsRequest(requests=formatted_report_requests[chunk_num_start:chunk_num_end], property='properties/' + self.property_id)).reports:
+                            raw_result['reports'].append(chunk_raw_res)
+            else:
+                raw_result = {}
+                raw_result['reports'] = self._api.batch_run_reports(BatchRunReportsRequest(requests=formatted_report_requests, property='properties/' + self.property_id)).reports
+
+            # We get back as raw_result:
+            #   { "reports" : [{ "columnHeader" : { "dimensions" : [Xh, Yh, Zh], "metricHeaderEntries" : [{ "name" : 1h, "type" : "INTEGER" }, ...] }, "data" : { "rows": [{ "dimensions" : [X,Y,Z], "metrics" : [1,2,3,4] }] }  }, { .. }, ....] }
+            raw_result['requests'] = formatted_report_requests
+            raw_result['report_key_names'] = report_key_names
+            # This transforms raw_result["reports"] into more usable data structure for ES and aggregation
+            #   e.g. list of JSON items instead of multi-dimensional table representation
+            return self.transform_report_result(
+                raw_result,
+                date_increment=kwargs.get('increment')
+            )
+
+
+
+        def get_latest_tracking_item_date(self, increment="daily"):
+            """
+            Queries '/search/?type=TrackingItem&sort=-google_analytics.for_date&&google_analytics.date_increment=...'
+            to get date of last TrackingItem for increment in database.
+
+            TODO: Accept yearly once we want to collect & viz it.
+            """
+            if increment not in ('daily', 'monthly'):
+                raise IndexError("increment parameter must be one of 'daily', 'monthly'")
+
+            search_results = ff_utils.search_metadata(
+                '/search/?type=TrackingItem&tracking_type=google_analytics&sort=-google_analytics.for_date&limit=1&google_analytics.date_increment=' + increment,
+                key=dict(self.owner.access_key, server=self.owner.server),
+                page_limit=1
+            )
+            if len(search_results) == 0:
+                return None
+
+            iso_date = search_results[0]['google_analytics']['for_date']
+
+            # TODO: Use date.fromisoformat() once we're on Python 3.7
+            year, month, day = iso_date.split('-', 2) # In python, months are indexed from 1 <= month <= 12, not 0 <= month <= 11 like in JS.
+            return date(int(year), int(month), int(day))
+
+
+
```

Comment:
extra spacing here and throughout, just use 1 space when within a class, flake8 should flag a lot of stuff you might want to consider fixing